format-code/action.yml

```diff
@@ -0,0 +1,83 @@
+name: Code Formatter
+description: "Runs MegaLinter to format code and apply linting rules across the repository."
+author: "Aaron Robinson"
+
+on:
+  workflow_call:
+    inputs:
+      flavor:
+        required: false
+        default: "full"
+        type: string
+      apply_fixes:
+        required: false
+        default: "all"
+        type: string
+      apply_fixes_event:
+        required: false
+        default: "all"
+        type: string
+      apply_fixes_mode:
+        required: false
+        default: "pull_request"
+        type: string
+      disable_errors:
+        required: false
+        default: "true"
+        type: string
+      email_reporter:
+        required: false
+        default: "false"
+        type: string
+      enable_linters:
+        required: false
+        default: "JSON_PRETTIER,YAML_PRETTIER,TERRAFORM_TERRAFORM_FMT,MARKDOWN_MARKDOWNLINT"
+        type: string
+      validate_all_codebase:
+        required: false
+        default: "true"
+        type: string
+      yaml_prettier_filter_regex_exclude:
+        required: false
+        default: "(.github/*)"
+        type: string
+      markdown_markdownlint_filter_regex_exclude:
+        required: false
+        default: ""
+        type: string
+      report_output_folder:
+        required: false
+        default: ""
+        type: string
+      ignore_files:
+        required: false
+        default: ""
+        type: string
+
+jobs:
+  format-code:
+    name: MegaLinter
+    runs-on: ubuntu-latest
+
+    steps:
+      - name: Checkout Code
+        uses: actions/checkout@11bd71901bbe5b1630ceea73d27597364c9af683 # v4.2.2
+        with:
+          fetch-depth: 0
+          token: ${{ secrets.GITHUB_TOKEN }}
+
+      - name: Run MegaLinter
+        uses: oxsecurity/megalinter/flavors/${{ inputs.flavor }}@e08c2b05e3dbc40af4c23f41172ef1e068a7d651 # v8.8.0
+        env:
+          APPLY_FIXES: ${{ inputs.apply_fixes }}
+          APPLY_FIXES_EVENT: ${{ inputs.apply_fixes_event }}
+          APPLY_FIXES_MODE: ${{ inputs.apply_fixes_mode }}
+          DISABLE_ERRORS: ${{ inputs.disable_errors }}
+          EMAIL_REPORTER: ${{ inputs.email_reporter }}
+          ENABLE_LINTERS: ${{ inputs.enable_linters }}
+          GITHUB_TOKEN: ${{ secrets.GITHUB_TOKEN }}
+          VALIDATE_ALL_CODEBASE: ${{ inputs.validate_all_codebase }}
```

Comment:
I'm not entirely sure about this (as I've not really tested it 🙈 ) - but when I did the Root Account Config, I wrote a dynamic value for the `VALIDATE_ALL_CODEBASE` i.e. Full Scan on `main` - changed files on a PR 🤹 Examples [here](https://github.com/ministryofjustice/aws-root-account/blob/3abadb8e1c8c4ed1d3041959797ad5401dcff550/.github/workflows/mega-linter.yml#L25) 🧪